apps/palette-generator/README.md

```diff
@@ -0,0 +1,141 @@
+# Palette Generator Web-App
+
+## How to Launch
+
+The palette generator is one of the web applications within this
+project that run on localhost.  It
+is made available using the following command executed in the home directory:
+
+```text
+npm run serveAppsDemos
+```
+
+Once the development server is running, open this `localhost` url from within a
+browser:
+[`http://localhost:5173/apps/palette-generator/palette-generator.html`](http://localhost:5173/apps/palette-generator/palette-generator.html)
+
+## How to Use
+
+### Settings
+
+The palette's settings are shown at the top of the page with the legend
+"Settings".  Within this block, use the "Palette name:" text field to enter a
+name for the palette.  The name can be any printable character including spaces.
+
+The two number fields "Starting row" and "Starting column" allow you to specify
+the top left corner of the palette.  The default values are row 1, column 1, and
+they refer to the position of the first cell in the first row of the palette.
+
+### Glosses
+
+The next section, "Glosses" is a text area where each line of text describes one
+row of the palette cells.  Spaces are used for separate cells so the text used
+to specify a cell cannot contain any spaces.
+
+There are three types of input allowed here:
+
+- a single word, such as "heart", "dog", or "singer",
+- a single number that is the BCI AV ID of a specific Bliss symbol,
+- the text "BLANK" will result in an empty blank cell,
+- an SVG string as required by the [bliss-svg-builder](https://github.com/hlridge/bliss-svg-builder)
+  that specifies component BCI AV IDs and shapes that the builder uses to create
+  the SVG graphic.  More about the structure of this string is given below.
+
+For the case of a single word or single BCI AV ID, the label to display can be
+given by immediately appending "LABEL:*label\_text*" to the word or number. The
+*label\_text* will be used as the label for the cell instead of the usual gloss
+for the symbol.  Any underscore characters in the *label\_text* will be replaced
+with a single space.   For example "clothLABEL:cross_hatches" will result in the
+the label "cross hatches"; and "23641LABEL:ruin" will give "ruin" as the label
+instead of the official gloss "ruin,wreck,wreckage\_(building)\_(1)".
+
+The SVG string is marked by "SVG:" at the beginning of the string and ":SVG" at
+the end. What lies between is a set of comma separated BCI AVI IDs, and quoted
+strings that are parsed by the svg builder.  If the string "LABEL:*label\_text"
+occurs immediately before the closing ":SVG" marker, it will be used as the
+label for the cell, and is truncated from the rest of the string before the
+remainder is sent to the builder.  The rules for the syntax of this "LABEL" is
```

Comment:
I tried with using "LABEL" with "SVG" marker. This string `SVG:14183,"/",25777,"/","W8W:0,8":SVGLABEL:aaa` leads to an error: "BciAvId not found for label: SVG:14183,"/",25777,"/","W8W:0,8":SVG". Did I combine the "SVG" and "LABEL" correctly?

Reply:
Quoting from the README with emphasis:
> If the string "LABEL:*label\_text" occurs _immediately before the closing ":SVG" marker_, 

So your example should be `SVG:14183,"/",25777,"/","W8W:0,8","LABEL:aaa":SVG`

However, that's different from the two other cases of an ID or a gloss-string, where the `LABEL:` is appended to the ID or string.  I think it should be the same regardless of the input.  I have re-written the processing so that your example works, and the old way is now illegal.